### PR TITLE
SCAL-81508-logo-and-footer-for-public-site

### PR DESCRIPTION
**Pull request by pankajp-ts**
Ticket - https://thoughtspot.atlassian.net/browse/SCAL-81508

**Desktop View:**
header-footer-desktop-view

**Mobile View:**
header-footer-mobile-view

### CODE REVIEW

File: docs/src/components/Footer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './index.scss';
+
+const Footer = () => (
+    <footer>
+        <section className="container">
+            <div className="footerWrapper">
+                <span>©2021 ThoughtSpot Inc. All Rights Reserved.</span>
```
Comment (ashubham) [inline docs/src/components/Footer/index.tsx:8]: do we have a strings file ?

File: docs/src/utils/app-utils.ts [hunk 2]
```diff
@@ -56,6 +61,9 @@ export const queryStringParser = (queryParamStr: string) => {
         PREVIEW_PREFIX
     ] = `${tsHostUrl}/#${queryParamObj[TS_APP_ROOT_PARAM]}`;
 
+    // used to check if the docs portal open as public site or inside embed iframe
+    queryParamObj[IS_PUBLIC_SITE_OPEN] = isPublicSiteOpen;
```
Comment (ashubham) [inline docs/src/utils/app-utils.ts:65]: Why are we putting this in queryParam ? This is not coming from query string in the url

**Review by ashubham — DISMISSED** [review]

File: docs/src/components/Footer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './index.scss';
+
+const Footer = () => (
+    <footer>
```
Comment (ashubham) [inline docs/src/components/Footer/index.tsx:5]: Is this approved by design ?

Comment (pankajp-ts) [reply docs/src/components/Footer/index.tsx:5]: I asked for designs but we don't have it as of now. Header is based on earlier design, footer is new.

File: docs/src/utils/app-utils.ts (hunk 2 above)
Comment (pankajp-ts) [reply docs/src/utils/app-utils.ts:65]: Actually this we are extracting by checking query params, that's why we put it there.

File: docs/src/components/Footer/index.tsx (hunk 1 above)
Comment (pankajp-ts) [reply docs/src/components/Footer/index.tsx]: Yes we have. Added all strings from all files.

File: docs/src/components/Footer/index.tsx (hunk 3 above)
Comment (ashubham) [reply docs/src/components/Footer/index.tsx:6]: Lets post this on the TSE channel to get feedback

File: docs/src/utils/app-utils.ts (hunk 2 above)
Comment (ashubham) [reply docs/src/utils/app-utils.ts:65]: But I dont think this should be here, maybe export another util method to check if its a public site ?

Comment (pankajp-ts) [reply docs/src/utils/app-utils.ts]: done

**Review by ShashiSubramanya — COMMENTED** [review]
Header looks good. 
You may want to revisit the footer design.

**Comment by harsh-ts** [comment]
@ashubham and @ShashiSubramanya please check new footer style

Desktop

image

Mobile

image

File: docs/src/utils/app-utils.ts [hunk 4]
```diff
@@ -63,4 +63,21 @@ export const removeTrailingSlash = (url: string) => {
...
+ * Used to check if the docs portal open as public site or inside embed iframe
+ * @param {string} queryParamStr - query string from location.search
+ * @returns {boolean} true, if it is opened as public site otherwise false
+ */
+export const isPublicSite = (queryParamStr: string) => {
+    const entries = new URLSearchParams(queryParamStr).entries();
+    let isPublicSiteOpen = true;
+    for (const [key, value] of entries) {
+        if ([TS_HOST_PARAM, TS_ORIGIN_PARAM, TS_APP_ROOT_PARAM].includes(key)) {
+            isPublicSiteOpen = false;
```
Comment (jbhanu-thoughtspot) [inline docs/src/utils/app-utils.ts:76]: this can be `return false`
Don't have to loop through all the entries.

File: docs/src/utils/app-utils.ts [hunk 5]
```diff
@@ -63,4 +63,21 @@ export const removeTrailingSlash = (url: string) => {
...
+        }
+    }
+
+    return isPublicSiteOpen;
```
Comment (jbhanu-thoughtspot) [inline docs/src/utils/app-utils.ts:80]: And here `return true`. Don't need to assign to a variable

**Review by jbhanu-thoughtspot — CHANGES_REQUESTED** [review]

**Review by jbhanu-thoughtspot — APPROVED** [review]
LGTM

**Pull request merged**